exporter/elasticsearchexporter/data_stream_router.go

```diff
@@ -91,3 +91,17 @@ func routeSpan(
 	route := routeWithDefaults(defaultDataStreamTypeTraces, defaultDataStreamDataset, defaultDataStreamNamespace)
 	return route(span.Attributes(), scope.Attributes(), resource.Attributes(), fIndex, otel)
 }
+
+// routeSpanEvent returns the name of the index to send the span event to according to data stream routing attributes.
+// This function may mutate record attributes.
```

Comment:
> This function may mutate record attributes.

I wonder if that could be an issue when using multiple exporters. Could this impact the records for other exporters so that there's a race condition where if the ES exporter for an event runs first, the other exporter also sees the mutated values?

I think we mutate records in the ES exporter in other places, too, so this isn't a new issue. If we think it could be a problem, let's create an issue and work towards making the ES exporter not mutate the records.

Reply:
> I think we mutate records in the ES exporter in other places, too, so this isn't a new issue. 

Agreed. I also discovered this potential issue while working on OTel mode. Let me create an issue to track this.

Reply:
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35023